### ISSUE TITLE
Accessing a form via index?

### ISSUE THREAD

**Issue by AleFons** [nathanvda/cocoon#397]
I have an array of data I want to input in a group of form fields, but I'm not sure how to actually access the form fields via an index - I've tried adding a field per position on the array, grabbing the index of the field via an after-insert bind, but then I don't know how to access the field with that index.

**Comment by nathanvda** [comment]
I guess I should be flattered that 1) you want to keep the message as short as possible to consume as little of my time as possible, 2) assume that this short/cryptic message makes any sense to me. 

Please show me some code, and explain what you tried and what you are trying to accomplish in a little more detail.  

Especially explain "access the field with that index" --> when? using javascript? you can already "grab the index of the field" ... so you already have the index? :confused:

**Comment by AleFons** [comment]
First, I must apologize; I made the post in a moment of frustration. I'll do my best to clear it up:

I'm using rails and javascript and I have multiple form fields of the same kind - set up in the model as a has_many. When the page loads, these fields are loaded with content normally.

However, the other thing is that I have an embedded googlemaps map on the page and I use the array of data from the form fields to draw on the map; however, when there's a change on the map, the array that represents the drawing changes and thus I want these changes to be reflected on the form fields.

My original idea for it was, whenever the array changed, I'd find all instances of the nested fields and delete them then make an amount of new forms equal to the length of the array and then populate the forms with data from it; at the same time, whenever a change to the nested fields happens, I have to change the array - so I need to catch the change events too.

This is the javascript function I used, but I did not figure out how to actually access the created form fields after creating them.

function redo_forms(){
  $('.remove_point').each(function() {
    $(this).trigger('click');
  });
  for(i=0; i < field_coords.length; i++){
    $('.add_point').trigger('click');
    //add data here
  };
}
